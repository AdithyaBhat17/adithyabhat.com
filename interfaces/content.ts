import { ResponsiveImageType, SeoMetaTagType } from 'react-datocms'

export type Article = {
  date: Date | string
  tags?: string
  slug?: string
  content?: string
  id?: string
  externalLink?: string
  thumbnail?: {
    alt: string
    blurhash: string
    title: string
    responsiveImage: ResponsiveImageType
  }
  seo?: SeoMetaTagType[]
  title: string
}

export type Project = {
  id: string
  title: string
  tags?: string
  externalLink?: string
  thumbnail?: {
    alt: string
    blurhash: string
    title: string
    responsiveImage: ResponsiveImageType
  }
  caseStudy: string
  date: Date | string
  slug: string
  link: string
  seo?: SeoMetaTagType[]
}

export interface ArticleProps {
  data: {
    article: Article
    moreArticles: Article[]
  }
}

export interface BlogPosts {
  data: {
    allArticles: Article[]
  }
}

export type WorkProps = {
  data: {
    allProjects: Project[]
  }
}

export type AllArticles = { allArticles: Article[] }
export type AllProjects = { allProjects: Project[] }

export interface List {
  data: AllArticles | AllProjects
  type: 'allProjects' | 'allArticles'
  columns: '2' | '3' | '4'
}
