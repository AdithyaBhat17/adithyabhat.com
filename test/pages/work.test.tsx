import { render } from '../testUtils'
import { cleanup } from '@testing-library/react'
import CaseStudy from '@/pages/work/[slug]'
import { mockProjects, mockProject } from '../__mocks__/content'
import Work from '@/pages/work'

afterAll(cleanup)

test('Renders a list of blogs', () => {
  const { container } = render(
    <Work
      data={{ allProjects: mockProjects.allProjects }}
      type="allArticles"
      columns="3"
    />
  )
  expect(container).toMatchSnapshot()
})

test('Renders a case study', () => {
  const { getByTestId } = render(
    <CaseStudy data={{ project: mockProject, allProjects: [] }} />
  )
  expect(getByTestId('title').textContent).toEqual('Homero')
  expect(getByTestId('content').textContent).toEqual('Homero case study')
})

test('Displays error page for invalid slug', () => {
  const { getByText } = render(
    <CaseStudy
      data={{ project: { ...mockProject, slug: '' }, allProjects: [] }}
    />
  )
  expect(getByText('404')).toBeInTheDocument()
})
