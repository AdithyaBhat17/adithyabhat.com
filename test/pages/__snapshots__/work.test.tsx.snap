// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a list of blogs 1`] = `
<div>
  <div>
    <nav
      class="navbar px-6 md:px-36 lg:px-48"
    >
      <div
        class="flex justify-start items-center"
      >
        <a
          class="hover:border-white"
          href="/"
        >
          <img
            alt="Adithya NR logo"
            class="fill-current lg:h-16 lg:w-16 w-12 h-12"
            src="/static/logo_black.svg"
          />
        </a>
        <span
          class="sm:inline hidden ml-6 text-gray-500 capitalize"
        />
      </div>
      <ul
        class="flex justify-between"
      >
        <li
          class="ml-6"
        >
          <a
            class=" text-black text-sm hover:border-gray-900"
            data-testid="nav-link"
            href="/about"
          >
            About
          </a>
        </li>
        <li
          class="ml-6"
        >
          <a
            class=" text-black text-sm hover:border-gray-900"
            data-testid="nav-link"
            href="/blog"
          >
            Blog
          </a>
        </li>
        <li
          class="ml-6"
        >
          <a
            class=" text-black text-sm hover:border-gray-900"
            data-testid="nav-link"
            href="/work"
          >
            Work
          </a>
        </li>
        <li
          class="ml-6"
        >
          <a
            class=" text-black text-sm hover:border-gray-900"
            data-testid="nav-link"
            href="/contact"
          >
            Contact
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="px-6 md:px-36 lg:px-48 mb-6"
      data-testid="container"
    >
      <h1
        class="font-semibold text-5xl text-gray-900 mb-6"
        style="opacity: 0; transform: translateY(60px) translateZ(0);"
      >
        Recent work
      </h1>
      <div
        class="flex items-start flex-wrap -mx-5"
      />
    </div>
    <br />
    <div
      class="px-6 md:px-36 lg:px-48 mb-6"
      data-testid="container"
    >
      <hr
        class="border-gray-800 mt-12 px-6"
      />
      <br />
      <footer
        class="footer flex-wrap"
      >
        <div
          class="flex justify-start items-center"
        >
          <a
            class="hover:border-white flex justify-start items-center"
            href="/"
          >
            <img
              alt="Adithya NR logo"
              class="fill-current lg:h-16 lg:w-12 w-12 h-12 mr-3"
              src="/static/logo_black.svg"
            />
            <span
              class="font-semibold text-black poppins"
            >
              adithya nr
            </span>
          </a>
        </div>
        <ul
          class="flex justify-between"
        >
          <li
            class="ml-0 md:ml-10"
          >
            <a
              class=" text-black text-sm hover:border-gray-900"
              data-testid="footer-link"
              href="/about"
            >
              About
            </a>
          </li>
          <li
            class="ml-0 md:ml-10"
          >
            <a
              class=" text-black text-sm hover:border-gray-900"
              data-testid="footer-link"
              href="/blog"
            >
              Blog
            </a>
          </li>
          <li
            class="ml-0 md:ml-10"
          >
            <a
              class=" text-black text-sm hover:border-gray-900"
              data-testid="footer-link"
              href="/work"
            >
              Work
            </a>
          </li>
          <li
            class="ml-0 md:ml-10"
          >
            <a
              class=" text-black text-sm hover:border-gray-900"
              data-testid="footer-link"
              href="/contact"
            >
              Contact
            </a>
          </li>
          <li
            class="ml-0 md:ml-10"
          >
            <a
              class=" text-black text-sm hover:border-gray-900"
              data-testid="footer-link"
              href="/resume.pdf"
            >
              Resume
            </a>
          </li>
          <li
            class="ml-0 md:ml-10"
          >
            <a
              class=" text-black text-sm hover:border-gray-900"
              data-testid="footer-link"
              href="/uses"
            >
              Uses
            </a>
          </li>
          <li
            class="ml-0 md:ml-10"
          >
            <a
              class=" text-black text-sm hover:border-gray-900"
              data-testid="footer-link"
              href="/legal"
            >
              Legal
            </a>
          </li>
        </ul>
        <div
          class="flex justify-start items-center mt-10 lg:mt-0"
        >
          <a
            class="text-black mr-5"
            data-testid="socials-link"
            href="https://behance.net/adithyabhat"
            rel="noreferrer noopener"
            target="_blank"
          >
            <img
              alt="behance"
              loading="lazy"
              src="/static/behance.svg"
            />
          </a>
          <a
            class="text-black mr-5"
            data-testid="socials-link"
            href="https://dribbble.com/adithyanr"
            rel="noreferrer noopener"
            target="_blank"
          >
            <img
              alt="dribbble"
              loading="lazy"
              src="/static/dribbble.svg"
            />
          </a>
          <a
            class="text-black mr-5"
            data-testid="socials-link"
            href="https://twitter.com/adithya__nr"
            rel="noreferrer noopener"
            target="_blank"
          >
            <img
              alt="twitter"
              loading="lazy"
              src="/static/twitter.svg"
            />
          </a>
          <a
            class="text-black mr-5"
            data-testid="socials-link"
            href="https://github.com/adithyabhat17"
            rel="noreferrer noopener"
            target="_blank"
          >
            <img
              alt="github"
              loading="lazy"
              src="/static/github.svg"
            />
          </a>
          <a
            class="text-black mr-5"
            data-testid="socials-link"
            href="https://linkedin.com/in/adithya-nr"
            rel="noreferrer noopener"
            target="_blank"
          >
            <img
              alt="linkedin"
              loading="lazy"
              src="/static/linkedin.svg"
            />
          </a>
        </div>
      </footer>
    </div>
  </div>
</div>
`;
